adapters/hono-adapter-amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -0,0 +1,382 @@
+        final Future<ProtonServer> result = Future.future();
+        final ProtonServer createdServer = (server == null) ? ProtonServer.create(this.vertx, options) : server;
+
+        createdServer.connectHandler(this::connectionRequestHandler).listen(res -> {
+            if (res.succeeded()) {
+                LOG.info(String.format("AMQP server running on %s:%d", getConfig().getInsecurePortBindAddress(),
+                        getConfig().getInsecurePort()));
+                result.complete(res.result());
+            } else {
+                result.fail(res.cause());
+            }
+        });
+        return result;
+    }
+
+    private void connectionRequestHandler(final ProtonConnection connRequest) {
+        LOG.info("Received connection request from client");
+        connRequest.disconnectHandler(conn -> {
+            LOG.error("Connection disconnected " + conn.getCondition().getDescription());
+        });
+        // when a BEGIN frame from a client
+        connRequest.sessionOpenHandler(session -> handleSessionOpen(connRequest, session));
+        // when an OPEN is received
+        connRequest.openHandler(remoteOpen -> {
+            ProtonConnection conn = remoteOpen.result();
+            conn.setContainer(getTypeName());
+            conn.open();
+        });
+        // when an Attach frame is received
+        connRequest
+                .receiverOpenHandler(remoteReceiverOpen -> handleRemoteReceiverOpen(remoteReceiverOpen, connRequest));
+        connRequest.senderOpenHandler(sender -> {
+            // this should not happen -> no request-response for device clients.
+            LOG.debug("client [container: {}] wants to open a link [address: {}] for receiving messages",
+                    connRequest.getRemoteContainer(), sender.getRemoteSource());
+            sender.setCondition(ProtonHelper.condition(AmqpError.NOT_ALLOWED,
+                    "this adapter only forwards message to downstream applications"));
+            sender.close();
+        });
+    }
+
+    /**
+     * Sets the AMQP server for handling insecure AMQP connections.
+     * 
+     * @param server The insecure server instance.
+     * @throws NullPointerException If the server is {@code null}.
+     */
+    protected void setInsecureAmqpServer(final ProtonServer server) {
+        Objects.requireNonNull(server);
+        if (server.actualPort() > 0) {
+            throw new IllegalArgumentException("AMQP Server should not be running");
+        } else {
+            this.insecureServer = server;
+        }
+    }
+    /**
+     * This method is called when an AMQP BEGIN frame is received from a remote client. This method sets the incoming
+     * capacity in its BEGIN Frame to be communicated to the remote peer
+     *
+     */
+    private void handleSessionOpen(final ProtonConnection conn, final ProtonSession session) {
+        LOG.debug(String.format("opening new session with client [container: {}]", conn.getRemoteContainer()));
```

Comment:
no need to use String.format in a logging statement. Logger.debug supports placeholder substitution natively, e.g.
`LOG.debug("opening new session with client [container: {}]", conn.getRemoteContainer());`
In fact, your statement does not work because String.format expects placeholders like `%s` for Strings or `%d` for Numbers